### PR TITLE
fix(deps): remove requirement of symfony/cache

### PR DESCRIPTION
**Pull request by Gounlaf**
Move symfony/cache dependency to dev dependencies because it's used only in tests.

### CODE REVIEW

**Comment by natebrunette** [comment]
https://github.com/tebru/doctrine-annotation-reader/blob/a9a60884ee447a43cda3e6243afc67e87179bc9d/src/AnnotationReaderAdapter.php#L41

**Comment by Gounlaf** [comment]
> https://github.com/tebru/doctrine-annotation-reader/blob/a9a60884ee447a43cda3e6243afc67e87179bc9d/src/AnnotationReaderAdapter.php#L41

Yes, `CacheInterface` from `psr/simple-cache` :wink: 

https://github.com/tebru/doctrine-annotation-reader/blob/a9a60884ee447a43cda3e6243afc67e87179bc9d/src/AnnotationReaderAdapter.php#L12

**Comment by Gounlaf** [comment]
I fixed tests;
Btw, minimum version should be upgrade to php>=7.2 [since php 7.1 is no longer supported](https://www.php.net/supported-versions.php)

**Comment by Gounlaf** [comment]
Hi @natebrunette,

Have you some times to review this PR?
I'm using _your retrofit version_, but unfortunately they all have some "fixed dependencies" for symfony components that lock upgrading to newer version =/

Thanks,
Regards,

**Comment by natebrunette** [comment]
Sorry! This fell off my radar. So the unit test changes seem unrelated, correct?

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
-/vendor/
-/build/
+vendor/
+build/
```
Comment (natebrunette) [inline .gitignore]: We only want to exclude directories in the project root named vendor or build

**Comment by Gounlaf** [comment]
> So the unit test changes seem unrelated, correct?

They are unrelated to the change for symfony/cache, but tests didn't pass; that's why if fixed them in a 2nd commit

File: .gitignore (hunk 1 above)
Comment (Gounlaf) [reply .gitignore]: Sorry, fixed

**Comment by natebrunette** [comment]
@Gounlaf thanks! Could you just squash these? I'll merge it and tag a new release.

**Comment by Gounlaf** [comment]
> @Gounlaf thanks! Could you just squash these? I'll merge it and tag a new release.

Squashed :)

**Pull request merged**